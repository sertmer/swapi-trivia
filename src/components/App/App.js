import React, { Component } from 'react';
import './App.scss';
import Header from '../Header/Header'
import LandingForm from '../LandingForm/LandingForm'
import MovieContainer from '../MovieContainer/MovieContainer'


class App extends Component {
  constructor() {
    super()
    this.state = {
      movies: [],
      name: '',
      quote: '',
      rank: ''
    }
  }
  
  componentDidMount() {
    fetch('https://swapi.co/api/films/')
      .then(res => res.json())
      .then(data => {
        let sortedMovies = data.results.sort((a, b) => a.episode_id - b.episode_id)
        this.setState({movies: sortedMovies})
      })
  }

  updateState = (statesObj) => {
    this.setState(statesObj);
  }

  signOut = () => {
    this.setState({ name: '', quote: '', rank: '' })
  }

  render() {
    return (
      <div className="App">

        <Header 
        name={this.state.name}
        rank={this.state.rank}
        quote={this.state.quote}
        signOut={this.signOut}
        />
        <LandingForm updateState={this.updateState}/>
        <MovieContainer 
          className='MovieContainer'
          movies={this.state.movies}
        />

      </div>
    )
  }
}

export default App;
