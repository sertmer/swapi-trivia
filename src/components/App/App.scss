* {
  box-sizing: border-box;
  font-family: 'Open Sans', sans-serif;
  margin: 0;
  padding: 0;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: black;
  background-image: url('https://wallpaperfm.com/img/original/0/d/a/16845.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  color: #f9d71c;
  min-height: 150vh;
  height: 100%;
  
}