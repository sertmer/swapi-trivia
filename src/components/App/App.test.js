import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { shallow } from 'enzyme'

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  
  it('renders without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<App />, div)
      ReactDOM.unmountComponentAtNode(div)
  })

  it('should match the snapshot', () => {   
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when updateState is called', () => {
    const expected = {
      movies: [],
      name: 'bob',
      quote: 'there is no try',
      rank: 'Jedi Master'
    }

    const defaultState = {
      movies: [],
      name: '',
      quote: '',
      rank: '',
    }

    wrapper.setState(defaultState)
    expect(wrapper.state()).toEqual(defaultState)

    wrapper.instance().updateState(expected)
    expect(wrapper.state()).toEqual(expected)

  })
})  
