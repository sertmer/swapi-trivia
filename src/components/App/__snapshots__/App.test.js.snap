// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Header
    name=""
    quote=""
    rank=""
    signOut={[Function]}
  />
  <LandingForm
    updateState={[Function]}
  />
  <MovieContainer
    className="MovieContainer"
    movies={Array []}
    updateState={[Function]}
  />
</div>
`;
